### ISSUE TITLE
Example of invalid spec passing

### ISSUE THREAD

**Issue by sargun** [anweiss/cddl#87]
```
sargun:oci2 sargun$ ./cddl-darwin-amd64 --version
cddl 0.8.5
sargun:oci2 sargun$ ./cddl-darwin-amd64 compile-cddl --cddl hello.cddl 
hello.cddl is conformant
```

The contents:

```cddl
; 1868785970 is "oci2" as an integer
oci2 = #6.1868785970({
    ; A given version of the spec may use a specific set
    ; of hash schemes, file layouts, etc.. Therefore in order
    ; to allow for multiple versions of the schema to exist
    ; simultaneously, a user can quickly read this as the
    ; basis of comparison.
    version: uint,
    files: {
        *filename => file,
    }
})

; Consider restricting this
filename = tstr

file = {
    mode: mode,
    (
        ? uid: unsigned,
        ? gid: unsigned //
        ? username: tstr,
        ? groupname: tstr
    )dwadawda
    ; Access time
    ? atime: tdate,
    ; Modification time
    ? mtime: tdate,
    content: content,
}

content = {
    $$content,
}

$$content //= (
    type: "regularfile",
    regularfile: [
        ; A 0 lengthed file may omit the hash.
        size: uint,
        ; Blake 3 256-bit hash
        ? b3-256: bstr .size 256,
        ; Because this is a vector, it would require
        ; revving the specification.
        ;
        ; TODO: Consider adding new hash types.
        ; TODO: Consider adding holes.
    ],
)

$$content //= (
    type: "directory",
    directory: []
)

$$content //= (
    type: "link",
    link: [
        target: tstr,
    ]
)

$$content //= (
    type: "symlink",
    symlink: [
        target: tstr,
    ]
)

$$content //= (
    type: "character",
    character: [
        major: uint .le 18446744073709551615,
        minor: uint .le 18446744073709551615,  
    ]
)

$$content //= (
    type: "block",
    block: [
        major: uint .le 18446744073709551615,
        minor: uint .le 18446744073709551615, 
    ]
)

$$content //= (
    type: "fifo",
    block: []
)

rwx = [
    read: bool,
    write: bool,
    execute: bool,
]

mode = [
    user: rwx,
    group: rwx,
    other: rwx,
    setuid: bool,
    setgid: bool,
    sticky: bool,
]

```

The part that's "dwadawda" is invalid.

**Comment by anweiss** [comment]
Thanks for reporting @sargun!

**Comment by anweiss** [comment]
@cabo, correct me if I'm wrong, but technically, the `dwadawda` is not invalid, but merely a group entry built from a not-yet-defined `dwadawda` named group. Granted, one would still consider this spec invalid from a readability standpoint pursuant to RFC 8160 Section 3.1:

> Except within strings, whitespace (spaces, newlines, and comments) is used to separate syntactic elements for readability (and to separate identifiers, range operators, or numbers that follow each other); it is otherwise completely optional.

**Comment by cabo** [comment]
```
$ echo "a = [0123]" | cddl - g
[0, 123]
```

**Comment by cabo** [comment]
Yes, the parsing may have some surprises, but I agree with @anweiss' diagnosis, except that an rule name that is not defined actually is an error except for a socket.

**Comment by cabo** [comment]
Time for the obfuscated cddl contest :-)

**Comment by anweiss** [comment]
@sargun FYI this is (finally) being fixed in #93 per @cabo's [comment](https://github.com/anweiss/cddl/issues/87#issuecomment-794175377)
